src/app/modules/consultations/consultation-list/consultation-list.component.html

```diff
@@ -72,13 +89,18 @@
             </div>
           </div>
           <div class="mt-3">
-            <button class="app-btn view-btn pointer" [routerLink]="['/consultations', item.id]">VIEW DISCUSSION</button>
+            <button *ngIf="item?.visibility?.toString() === 'public_consultation'" class="app-btn view-btn pointer"
+              [routerLink]="['/consultations', item.id]">VIEW DISCUSSION</button>
+            <div *ngIf="item?.visibility?.toString() === 'private_consultation'">
```

Comment:
@Harshwardhan18 You've assumed here that the only way someone will land on a Private Consultation page is through the Consultation listing page on Civis. What if I found the link in my email or if I typed the link directly into my browser? This approach would fail in that case. 
The correct way to do this would be to check the API response instead. If a user is unauthenticated, they cannot view the private consultation. The API will respond with a 401 itself if the access_token is not sent while requesting for a private consultation. That's how you should redirect them to auth. Not like this. Please change

Reply:
@avikt bro so I should remove the logic applied on buttons rather solely depend on consulationProfle component validation_check. As first the user will go to that page and then redirects upon error token while with button in place like this he'll go directly to auth private page

Reply:
@Harshwardhan18 Remove this logic. Only have it in the consultationProfile component.

Reply:
@Harshwardhan18 You didn't get rid of this? Please remove this. Private Consultations won't even appear in the Consultations List page

Reply:
@Harshwardhan18 Remove this